src/method.jl

```diff
@@ -0,0 +1,114 @@
+"""
+    signature(meth::Method) -> Dict{Symbol,Any}
+
+Finds the expression for a methods signature as broken up into its various components
+including:
+
+- `:name`: Name of the function
+- `:params`: Parametric types defined on constructors
+- `:args`: Positional arguments of the function
+- `:whereparams`: Where parameters
+
+All components listed above may not be present in the returned dictionary
+
+Right now the following components will never be returned:
+ - `:kwargs`: Keyword arguments of the function
+ - `:rtype`: Return type of the function
+ - `:body`: Function body
+
+These are the same components returned by [`splitdef`](@ref) and required by
+[`combinedef`](@red), except for the `:body` component which will never be present.
+"""
+function signature(meth::Method)
+    def = Dict{Symbol,Any}()
+    def[:name] = meth.name
+
+    def[:args] = get_args(meth)
+    def[:whereparams] = get_whereparams(meth)
+
+    return Dict(k=>v for (k, v) in def if v !== nothing)  # filter out nonfields.
+end
+
+function slot_syms(meth::Method)
+    # In 1.3 can do:
+    # Symbol.(split(meth.slot_syms, '\0'; keepempty=false))
+    ci = Base.uncompressed_ast(meth)
+    return ci.slotnames
+end
+function argument_names(meth::Method)
+    slot_syms = get_slot_sym(meth)
+    @assert slot_syms[1] == Symbol("#self#")
```

Comment:
Symbols are singletons right?
```suggestion
    @assert slot_syms[1] === Symbol("#self#")
```